File: src/main/java/CustomCellRenderer.java
import javax.swing.*;
import java.awt.*;
import java.util.Hashtable;

public class CustomCellRenderer implements ListCellRenderer {
    DefaultListCellRenderer listCellRenderer = new DefaultListCellRenderer();

    public Component getListCellRendererComponent(JList list, Object value, int index, boolean selected, boolean hasFocus) {
        listCellRenderer.getListCellRendererComponent(list, value, index, selected, hasFocus);
        listCellRenderer.setText(getValueString(value));
        return listCellRenderer;
    }

    private String getValueString(Object value) {
        String returnString = "null";
        if (value != null) {
            if (value instanceof Hashtable) {
                Hashtable hashtable = (Hashtable) value;
                String name = (String) hashtable.get("name");
                String path = (String) hashtable.get("path");
                returnString = name + " ==> " + path;
            } else {
                returnString = "X: " + value;
            }
        }
        return returnString;
    }
}

File: src/main/kotlin/CustomCellRenderer.kt
package de.griefed

import java.awt.Component
import java.util.*
import javax.swing.DefaultListCellRenderer
import javax.swing.JList
import javax.swing.ListCellRenderer

class CustomCellRenderer : ListCellRenderer<Any> {
    var listCellRenderer: DefaultListCellRenderer = DefaultListCellRenderer()

    override fun getListCellRendererComponent(
        list: JList<out Any?>?,
        value: Any?,
        index: Int,
        isSelected: Boolean,
        cellHasFocus: Boolean
    ): Component {
        listCellRenderer.getListCellRendererComponent(list, value, index, isSelected, cellHasFocus)
        listCellRenderer.setText(getValueString(value))
        return listCellRenderer
    }

    private fun getValueString(value: Any?): String {
        var returnString = "null"
        if (value != null) {
            if (value is Hashtable<*, *>) {
                val hashtable = value
                val name = hashtable["name"] as String?
                val path = hashtable["path"] as String?
                returnString = "$name ==> $path"
            } else {
                returnString = "X: $value"
            }
        }
        return returnString
    }
}